data/routers/cohorts-router.js

```diff
@@ -0,0 +1,173 @@
+        .select('cohorts.name as Cohort', 'students.name as Student Name')
+        .where({ cohort_id: id });
+      res.status(200).json(cohortStudents);
+    } catch (err) {
+      res.status(500).json({
+        error: `There was an error while retrieving the cohort's students data. ${err}`
+      });
+    }
+  } else {
+    res.status(404).json({
+      error: 'There is no cohort with the specified ID.'
+    });
+  }
+});
+
+router.put('/:id', async (req, res) => {
+  const { id } = req.params;
+  const { name } = req.body;
+  if (!name) {
+    res.status(400).json({
+      error: 'Please provide the cohort changes you intend to make.'
+    });
+  } else {
+    try {
+      const updatedCohort = await db('cohorts')
+        .where({ id })
+        .update(req.body);
+      if (updatedCohort) {
+        try {
+          const updatedRecord = await db('cohorts')
+            .where({ id })
+            .first();
+          res.status(200).json(updatedRecord);
+        } catch (err) {
+          res.status(500).json({
+            error: `There was an error while retrieving the updated cohort data. ${err}`
+          });
+        }
+      } else {
+        res.status(404).json({
+          message: 'There is no cohort with the specified ID.'
+        });
+      }
+    } catch (err) {
+      res.status(500).json({
+        error: `There was an error while updating the cohort data. ${err}`
+      });
+    }
+  }
+});
+
+router.delete('/:id', async (req, res) => {
+  const { id } = req.params;
+  try {
+    const cohort = await db('cohorts')
+      .where({ id })
+      .first();
+    if (cohort) {
+      try {
+        const cohortStudents = await db('cohorts')
+          .join('students', 'cohorts.id', 'students.cohort_id')
+          .select('students.id')
+          .where({ cohort_id: id });
+        if (cohortStudents.length > 0) {
+          res.status(400).json({
+            error:
+              'You may not delete a cohort which still has students assigned to it.'
+          });
+        } else {
+          try {
+            const numOfDeletedCohorts = await db('cohorts')
+              .where({ id })
+              .del();
+            if (numOfDeletedCohorts) {
+              res.status(200).json({
+                message: `Number of cohorts deleted: ${numOfDeletedCohorts}.`
+              });
+            }
+          } catch (err) {
+            res.status(500).json({
+              error: `There was an error while deleting the cohort. ${err}`
+            });
+          }
+        }
+      } catch (err) {
+        res.status(500).json({
+          error: `There was an error while checking whether the cohort has any students assigned to it. ${err}`
+        });
+      }
+    } else {
+      res.status(404).json({
+        error: 'There is no cohort with the specified ID.'
+      });
+    }
+  } catch (err) {
+    res.status(500).json({
+      error: `There was an error while checking the cohort ID. ${err}`
+    });
+  }
```

Comment:
craaazy amount of cases to check for here the way you're doing it--I'm assuming you're adding the extra conditionals and checking for the existence of a cohort's students before cohort deletion because you starting running into an error when seeing if that would work, even though the .onDelete('CASCADE')` in your student migrations should have taken care of that, and there's some strange behavior with SQLite that requires you to actually _turn on_ your FKs after the database is created: this can be done by adding an extra property in your knexConfig object. I don't think Luis went over it explicitly for lecture, I showcase it real quick at the start of our standup tonight

Reply:
` pool: {
    afterCreate: (conn, cb) => conn.run("PRAGMA foreign_keys = ON", cb)
  }` but you "turn on" the FKs by adding this in your knexConfig